Bug 1421664 Converting Primer to Getting Started Guide

bmcelvee opened the pull request:
https://bugzilla.redhat.com/show_bug.cgi?id=1421664

bmcelvee commented:
openshift-enterprise: http://file.rdu.redhat.com/~bmcelvee/06042017/Primer-to-GSG/getting_started/
openshift-origin: http://file.rdu.redhat.com/~bmcelvee/06042017-1/Primer-to-GSG/getting_started/
openshift-dedicated: http://file.rdu.redhat.com/~bmcelvee/06042017-2/Primer-to-GSG/getting_started/
openshift-online: http://file.rdu.redhat.com/~bmcelvee/06042017-3/Primer-to-GSG/getting_started/

bmcelvee commented:
@mburke5678 peer review please 😸 

@aheslin Something went wonky with the old PR/branch. This one is up to date and the preview files are linked.

mburke5678 commented on getting_started/index.adoc:
```diff
@@ -8,8 +8,33 @@
+[[introduction]]
+== Introduction
+{product-title} is a platform as a service (PaaS) offering from Red Hat
+that brings together Docker, Kubernetes, and an API to manage these services.
```
The API manages docker and kube? If so, maybe: brings together Docker and Kubernetes, and provides and API to manage these services.

mburke5678 commented on getting_started/index.adoc:
```diff
@@ -8,8 +8,33 @@
+independent of operating system and the underlying infrastructure. OpenShift
+helps you to develop, deploy, and manage container-based applications. It
+provides you with a self-service platform to create, modify, and deploy
+applications on demand. Thus enabling faster development and release life
```
A *real* quick Google search seems to show that "thus enabling" is a phrase (or clause or whatever). 
"...deploy applications on demand, thus enabling..." Does that look right?

mburke5678 commented:
@bmcelvee Couple of little things. Otherwise LGTM.

mburke5678 commented:
@bmcelvee Couple of little things. Otherwise LGTM.

mburke5678 commented on getting_started/index.adoc:
```diff
@@ -62,3 +87,79 @@ For the {product-title} (Next Gen) Developer Preview offering, consult the
+* xref:../getting_started/beyond_the_basics.adoc#deploy-docker-registry[a Docker Registry]
+
+[[install-openshift]]
+== Install {product-title}
```
@vikram-redhat @bmcelvee @aheslin If the goal is to describe how to install a non-production instance, I liked the single-page organization of the primer. One page made the process very easy to follow and stay on-task.I didn't make this comment as I didn't know the intent of the Getting Started.

vikram-redhat replied:
@mburke5678 - hey Michael, yeah the intent is to guide users with a non production instance.

bmcelvee commented:
@openshift/team-documentation PTAL 

The formatting is off in this version of my preview file, but it's rendering normally when I build locally: http://file.rdu.redhat.com/~bmcelvee/25_07_2017/getting_started/index.html

bfallonf commented on getting_started/configure_openshift.adoc:
```diff
@@ -0,0 +1,237 @@
+----
++
+Note that you have created a
+user with the username of `admin` and password of `redhat`.
```
It might be more accurate to link the user and their password more. Maybe something like "...you have created a user named `admin` with a password of `redhat`."

bfallonf commented on getting_started/configure_openshift.adoc:
```diff
@@ -0,0 +1,237 @@
++
+----
+$ oadm router --replicas=1 --service-account=router
+----
```
Does it say anywhere in the docs why we delete the default and create a new one straight away? Is it just for this quick demo install? Should it say all this here? I'd imagine the reader would think this is weird (I do too).

bfallonf commented on getting_started/configure_openshift.adoc:
```diff
@@ -0,0 +1,237 @@
+----
+
+. Create the *docker-registry* service in the *default* project using the
+*registry* service account:
```
A bit of a nit, but the above ends with a colon but a lot of the steps in previous procedures just end with periods. Change for consistency?

bfallonf commented on getting_started/configure_openshift.adoc:
```diff
@@ -0,0 +1,237 @@
+=== Provision the Persistent Volume
+
+. Create a registry volume file on your master, as shown here, and call it
+*_registry-volume.yaml_*;
```
s/;/:

bfallonf commented on getting_started/index.adoc:
```diff
@@ -8,26 +8,56 @@
+(PaaS) offering from Red Hat that brings together Docker and Kubernetes, and
+provides and API to manage these services. {product-title} allows you to create
+and manage containers. Containers are standalone processes that run within their
+own environment.
```
This sentence is repeated on the next line (l24). Maybe get rid of this one? The next has a little more context.

bfallonf commented on getting_started/install_openshift.adoc:
```diff
@@ -0,0 +1,229 @@
+----
+====
+
+Once you have acquired these items, use these steps to set up a two-machine
```
Not sure if "acquired these items" is the best phrasing. Maybe "Once these are configured" or something similar?

bfallonf commented on getting_started/install_openshift.adoc:
```diff
@@ -0,0 +1,229 @@
+$ subscription-manager list --available
+----
+
+. Find the pool ID that provides {product-title} Subscription and attach it:
```
s/Subscription/subscription (case)

bfallonf commented:
@bmcelvee I made some comments. Some might be prefernces... LGTM overall!

vikram-redhat commented:
Hey @bmcelvee - were you planning on including the sections on the router/registry/storage? Not sure if you have discussed this with @aheslin ?

mburke5678 commented on getting_started/install_openshift.adoc:
```diff
@@ -0,0 +1,229 @@
+To install {product-title}, you will need:
+
+* At least two physical or virtual RHEL 7+ machines, with fully qualified domain
+names (either real world or within a network) and password-less ssh access to
```
s/ssh/SSH ??

mburke5678 commented on getting_started/install_openshift.adoc:
```diff
@@ -0,0 +1,229 @@
+* At least two physical or virtual RHEL 7+ machines, with fully qualified domain
+names (either real world or within a network) and password-less ssh access to
+each other. This guide, uses master.openshift.example.com and
+node.openshift.example.com. These machines must be able to ping each other
```
Should these machine names have mark-up?

mburke5678 commented on getting_started/install_openshift.adoc:
```diff
@@ -0,0 +1,229 @@
+----
+*.cloudapps.openshift.example.com. 300 IN  A <IP of the node>
+----
+====
```
Should the codeblock and "Why the apps..." section be tucked under the bullet above?

bmcelvee commented:
@vikram-redhat Yes, those sections are in the Configure section. 

http://file.rdu.redhat.com/%7Ebmcelvee/25_07_2017/getting_started/configure_openshift.html

mburke5678 commented on getting_started/install_openshift.adoc:
```diff
@@ -0,0 +1,229 @@
+====
+Instead of `rhel-7-server-ose-3.6-rpms`, use `rhel-7-server-ose-3.5-rpms` for
+OpenShift Container Platform 3.5 and `rhel-7-server-ose-3.4-rpms` for OpenShift
+Container Platform 3.4
```
This could be a bit stronger. 
"If you are using OpenShift Container Platform 3.5 use `rhel-7-server-ose-3.5-rpms`.
If you are using OpenShift Container Platform 3.4 use `rhel-7-server-ose-3.4-rpms`.

mburke5678 commented on getting_started/install_openshift.adoc:
```diff
@@ -0,0 +1,229 @@
+
+[[set-up-password-less-ssh]]
+=== Set up Password-less SSH Access
+
```
Haven't I already done this as a prereq?

bmcelvee replied:
Yes, thank you. I added a link to clarify this step.

mburke5678 commented on getting_started/install_openshift.adoc:
```diff
@@ -0,0 +1,229 @@
+option for {product-title}.
+Also, you are installing one master and one node and the domain name is the FQDN
+as mentioned at the start of this section (master.openshift.example.com and
+node.openshift.example.com).
```
mark-up on machine names?

mburke5678 commented on getting_started/install_openshift.adoc:
```diff
@@ -0,0 +1,229 @@
+----
+systemctl start atomic-openshift-master
+----
+
```
Do we have the output for a successful install? troubleshooting for failed install?

mburke5678 commented on getting_started/install_openshift.adoc:
```diff
@@ -0,0 +1,229 @@
+make this change yourself by looking for the `subdomain` entry.
+
+This install process takes approximately 5-10 minutes.
+
```
Do we have the output for a successful install? troubleshooting for failed install?

mburke5678 commented on getting_started/install_openshift.adoc:
```diff
@@ -0,0 +1,229 @@
+*system:admin* user's configuration file, which will allow you to log in
+subsequently.
+
+There is no password for this system account.
```
Do we want a Next Step link here?

vikram-redhat commented:
Ah. Thanks @bmcelvee! Somehow I missed that.

vikram-redhat commented:
Hey Brandi - this looks good to me. It might need some statements at the end of the installation steps to go to configuration topic and at the start of the configuration topic to let them know that they are following on from the basic non production installation steps. 

Otherwise, all good. Thanks!! :+1:

bmcelvee commented:
[rev_history]
|xref:../getting_started/index.adoc#getting-started-index[Overview]
|Added xref:../getting_started/index.adoc#introduction[Introduction] and xref:../getting_started/index.adoc#why-use-openshift[Why should I use OpenShift?] sections.
%
|xref:../getting_started/configure_openshift.adoc#getting-started-configure-openshift[Configure {product-title}]
|New topic descibing how to configure a basic OpenShift environment.
%
|xref:../getting_started/install_openshift.adoc#getting-started-install-openshift[Install {product-title}]
|New topic descibing how to install a basic OpenShift environment.
%
|xref:../getting_started/developers_cli.adoc#getting-started-developers-cli[Create and Build an Image Using the CLI]
|Updated topic title and removed link to outdated video.
%
|xref:../getting_started/developers_cli.adoc#getting-started-developers-console[Create and Build an Image Using the Web Console]
|Updated topic title and removed link to outdated video.
%

adellape commented on getting_started/index.adoc:
```diff
@@ -8,26 +8,55 @@
+and manage containers.
+
+[[why-use-openshift]]
+=== Why should I use OpenShift?
```
@bmcelvee Sorry if this came up in a previous comment and I missed: should this heading be title case? I.e., "Why Should I Use OpenShift?"

bmcelvee replied:
@adellape Ahh good catch. Opening a follow-up PR to fix.
